src/main/java/chess/repository/RoomRepositoryImpl.java

```diff
@@ -1,63 +1,87 @@
 import org.springframework.stereotype.Repository;
 
 @Repository
 public class RoomRepositoryImpl implements RoomRepository {
 
-    private static final String TABLE_NAME = "room";
-    private static final String KEY_NAME = "id";
-    private static final int KEY_INDEX = 1;
-    private static final int NAME_INDEX = 2;
-
-    private final SimpleJdbcInsert insertActor;
-    private final NamedParameterJdbcTemplate jdbcTemplate;
-
-    public RoomRepositoryImpl(DataSource dataSource,
-                              NamedParameterJdbcTemplate jdbcTemplate) {
-        this.insertActor = new SimpleJdbcInsert(dataSource)
-                .withTableName(TABLE_NAME)
-                .usingGeneratedKeyColumns(KEY_NAME);
-        this.jdbcTemplate = jdbcTemplate;
-    }
-
-    @Override
-    public int save(String name) {
-        return insertActor.executeAndReturnKey(Map.of("name", name)).intValue();
-    }
-
-    @Override
-    public Optional<RoomDto> find(String name) {
-        String sql = "select * from room where name = :name";
-        try {
-            return Optional.ofNullable(
-                    jdbcTemplate.queryForObject(sql, Map.of("name", name),
-                            (resultSet, rowNum) ->
-                                    new RoomDto(resultSet.getInt(KEY_INDEX), resultSet.getString(NAME_INDEX))
-                    ));
-        } catch (EmptyResultDataAccessException exception) {
-            return Optional.empty();
-        }
-    }
-
-    @Override
-    public Optional<RoomDto> findById(int roomId) {
-        String sql = "select * from room where id = :roomId";
-        try {
-            return Optional.ofNullable(
-                    jdbcTemplate.queryForObject(sql, Map.of("roomId", roomId),
-                            (resultSet, rowNum) ->
-                                    new RoomDto(resultSet.getInt(KEY_INDEX), resultSet.getString(NAME_INDEX))
-                    ));
-        } catch (EmptyResultDataAccessException exception) {
-            return Optional.empty();
-        }
-    }
+	private static final String TABLE_NAME = "room";
+	private static final String KEY_NAME = "id";
+	private static final int KEY_INDEX = 1;
+	private static final int NAME_INDEX = 2;
+	private static final int PASSWORD_INDEX = 3;
+
+	private final SimpleJdbcInsert insertActor;
+	private final NamedParameterJdbcTemplate jdbcTemplate;
+
+	public RoomRepositoryImpl(DataSource dataSource,
+		NamedParameterJdbcTemplate jdbcTemplate) {
+		this.insertActor = new SimpleJdbcInsert(dataSource)
+			.withTableName(TABLE_NAME)
+			.usingGeneratedKeyColumns(KEY_NAME);
+		this.jdbcTemplate = jdbcTemplate;
+	}
+
+	@Override
+	public int save(RoomDto roomDto) {
+		return insertActor.executeAndReturnKey(Map.of(
+				"name", roomDto.getName(),
+				"password", roomDto.getPassword()))
+			.intValue();
+	}
+
+	@Override
+	public Optional<RoomDto> find(String name) {
+		String sql = "select * from room where name = :name";
```

Comment:
```suggestion
		String sql = "select * from room where name = :name LIMIT 1";
```
1건 찾는 쿼리에는 LIMIT 1을 넣어주면 성능상 이점을 얻을 수 있을 것 같아요~